src/host.cpp

```diff
@@ -1084,18 +1109,13 @@ namespace rhost {
             transport::disconnected.connect(unblock_message_loop);
 
 #ifdef _WIN32
-            rp.ReadConsole = R_ReadConsole;
-            rp.WriteConsoleEx = WriteConsoleEx;
-            rp.CallBack = CallBack;
-            rp.ShowMessage = ShowMessage;
-            rp.YesNoCancel = YesNoCancel;
-            rp.Busy = Busy;
-
+            setCallbacksWindows(rp);
             char* dllVersion = getDLLVersion();
+            send_notification("!Microsoft.R.Host", 1.0, dllVersion);
 #else
-            char* dllVersion = "";
+            setCallbacksPOSIX();
+            send_notification("!Microsoft.R.Host", 1.0, "3.2.3");
```

Comment:
We'll need to figure out how to get the version right. But that can be a separate change.